src/Workspaces/Core/Portable/Storage/SQLite/v1/Interop/SafeSqliteHandle`1.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.CodeAnalysis.SQLite.v1.Interop
+{
+    internal abstract class SafeSqliteHandle<T> : SafeHandle
+        where T : class
+    {
+        protected readonly T? Wrapper;
+
+        public SafeSqliteHandle(IntPtr handle, T? wrapper)
+            : base(invalidHandleValue: IntPtr.Zero, ownsHandle: true)
+        {
+            Wrapper = wrapper;
+            SetHandle(handle);
+        }
+
+        public override bool IsInvalid => handle == IntPtr.Zero;
+
+        public new T DangerousGetHandle()
+            => Wrapper!;
```

Comment:
is htere a way to access the current ref-count of a SafeHandle (even though it would be dangerous of course).  If we can, ew could at least assert our ref count should be at least 1 here.

Reply:
`Dangerous*` methods typically do not include additional checks. This pattern follows the same one established by the base type.